src/hotspot/share/utilities/debug.cpp

```diff
@@ -732,7 +732,7 @@ static void store_context(const void* context) {
   *((void**) &g_stored_assertion_context.uc_mcontext.regs) = &(g_stored_assertion_context.uc_mcontext.gp_regs);
 #elif defined(AMD64)
   // In the copied version, fpregs should point to the copied contents. Preserve the offset.
-  intptr_t fpregs_offset = (address)(void*)(((const ucontext_t*)context)->uc_mcontext.fpregs) - (address)context;
+  size_t fpregs_offset = pointer_delta(((const ucontext_t*)context)->uc_mcontext.fpregs, context, 1);
```

Comment:
Interesting note, above I see we do similar adjustments for ppc (https://github.com/openjdk/jdk/commit/3e603a776ea9c5642cb0ec6b9105c6ff34b8f2b1)